cleanup/tests/test_cli.py

```
@@ -0,0 +1,129 @@
+    r = _run(["--mode", "rules"], stdin="")
+    assert r.returncode == 0
+    assert r.stdout == ""
+
+
+def test_cli_llm_falls_back_to_rules_when_daemon_missing_no_spawn():
+    # With --no-spawn, the CLI must not auto-spawn the daemon; LLM fails;
+    # the output should be the rules-cleaned version.
+    r = _run([
+        "--mode", "llm",
+        "--no-spawn",
+        "--socket", "/tmp/macparakeet-cleanup-nonexistent-xyz.sock",
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/macparakeet-cleanup-nonexistent-xyz.sock"

(S108)